### PR TITLE
Feature/sort modules by name

### PR DESCRIPTION
**Pull request by PedroErnst**
Make it possible for users to select alphabetical order for the modules in the dropdown lists in the top navigation menu.

## Description

Adds a checkbox in Profile to enable alphabetical ordering.

## Motivation and Context

The module list is unordered by default, and in the list with ALL modules a module can be hard to find.
In addition, the current module gets moved to an unpredictable position (usually around the 10th in the list) and that's awkward.

screenshot from 2017-12-29 12-21-16

Added a checkbox in the user profile to enable alphabetical sorting.

screenshot from 2017-12-29 13-38-03

Now the list of modules is nicely sorted.
In addition the current module is always on top of the list (this happens even if alphabetical sorting is disabled).

screenshot from 2017-12-29 12-20-46

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

**Comment by horus68** [comment]
Excellent!
Maybe also include changes in Module names when they are related so they can be listed next to weach other.
See related request: https://github.com/salesagility/SuiteCRM/pull/1866

**Comment by PedroErnst** [comment]
Yes that's great, makes it all the easier to find related modules.

**Comment by PedroErnst** [comment]
@Dillon-Brown merge conflicts are solved

**Comment by Dillon-Brown** [comment]
Assessed

File: modules/Users/Save.php [hunk 1]
```diff
@@ -199,6 +199,12 @@
         $focus->setPreference('navigation_paradigm', 'gm', 0, 'global');
     }
 
+    if (isset($_POST['sort_modules_by_name'])) {
+        $focus->setPreference('sort_modules_by_name', $_POST['sort_modules_by_name'], 0, 'global');
+    } else {
+        $focus->setPreference('sort_modules_by_name', $_POST['sort_modules_by_name'], 0, 'global');
```
Comment (samus-aran) [inline modules/Users/Save.php]: Eh? I believe this should be '' rather?

**Pull request merged**